From 8a2936a4763c3ea6594a71aec1300cd45053f2e0 Mon Sep 17 00:00:00 2001
From: Noriko Hosoi <nhosoi@redhat.com>
Date: Mon, 21 Mar 2011 16:44:16 -0700
Subject: [PATCH 5/7] Bug 681015 - RFE: allow fine grained password policy duration attributes in days, hours, minutes, as well

https://bugzilla.redhat.com/show_bug.cgi?id=681015

Description: passwordLockoutDuration attribute is not working
with the fine grain password policy.  The code to parse the
value of passwordlockoutduration was missing.  This patch
adds it.
---
 ldap/servers/slapd/pw.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/ldap/servers/slapd/pw.c b/ldap/servers/slapd/pw.c
index 2992623..7c6210b 100644
--- a/ldap/servers/slapd/pw.c
+++ b/ldap/servers/slapd/pw.c
@@ -1673,7 +1673,7 @@ new_passwdPolicy(Slapi_PBlock *pb, char *dn)
 				else
 				if (!strcasecmp(attr_name, "passwordlockoutduration")) {
 					if ((sval = attr_get_present_values(attr))) {
-						pwdpolicy->pw_lockduration = slapi_value_get_long(*sval);
+						pwdpolicy->pw_lockduration = slapi_value_get_timelong(*sval);
 					}
 				}
 				else
-- 
1.7.1

